### PR TITLE
added pin note button in the document-bar

### LINKED ISSUE #331: Add 'pin'/'favorite' button in the note editor
Currently, it is possible to 'pin' a note in the note history view.
This feature should also be available in the note editor itself.
It seems counter-intuitive to read a note, find it 'pin worthy' and then need to switch to the history view to actually pin it.

The easiest way to add this is probably to add a pin (or favorite star) button in the top taskbar.

This is blocked by #330

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
pin button in editor

### Description
This PR add the functionality to pin notes from the editor.

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [x] extended changelog

### Related Issue(s)
closes #331

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@
 - Asciinema videos may now be embedded by pasting the URL of one video into a single line
 - The Toolbar includes an EmojiPicker
 - Added shortcodes for [fork-awesome icons](https://forkaweso.me/Fork-Awesome/icons/) (e.g. `:fa-picture-o:`)
+- Users can pin a note in the editor.
```
Comment (ErikMichelson) [inline CHANGELOG.md:39]: Clarify, that the note will be pinned in the history not in the editor itself.

File: src/api/history/index.ts [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+import { defaultFetchConfig, expectResponseCode, getApiUrl } from '../utils'
 
 export const getHistory = async (): Promise<HistoryEntry[]> => {
   const response = await fetch(getApiUrl() + '/history')
   expectResponseCode(response)
   return await response.json() as Promise<HistoryEntry[]>
 }
 
+export const getHistoryEntry = async (noteId: HistoryEntry['id']): Promise<HistoryEntry> => {
+  const response = await fetch(getApiUrl() + '/history' + noteId)
```
Comment (ErikMichelson) [inline src/api/history/index.ts:11]: This looks as if a slash is missing or does the noteId start with one?

File: CHANGELOG.md (hunk 1 above)
Comment (DerMolly) [reply CHANGELOG.md]: done

File: src/api/history/index.ts (hunk 2 above)
Comment (DerMolly) [reply src/api/history/index.ts]: nope this was a mistake

Comment (DerMolly) [reply src/api/history/index.ts]: done

**Comment by DerMolly** [comment]
I discussed this with @InnayTool and we both think that the functionality of this button needs to be implemented via the websocket. Therefore I'll close this PR for now and work on it again if the websocket is implemented.

**Comment by mrdrogdrog** [comment]
What about the idea with the redux state? We could build that now.

**Comment by mrdrogdrog** [comment]
Or the refactoring of the history pin button?

**Pull request closed without merging**